### PR TITLE
Preview

### PR DESCRIPTION
**Pull request by faethonm**

### CODE REVIEW

**Comment by cionescu** [comment]
LGTM

https://i.imgur.com/vUJBo79.gif

File: app/models/answer.rb [hunk 1]
```diff
@@ -24,4 +24,7 @@ class Answer < ApplicationRecord
   # def complete?
   #   candidate_application.answers.all?{|a| a.content.present?}
   # end
+
+scope :code_answers, -> {joins(:question).where('questions.type = ?', 'CodeQuestion')}
```
Comment (cionescu) [inline app/models/answer.rb]: Please indent this line, add a space after `{` and you can write it in a less sql way:
`where(questions: { type: "CodeQuestion" })`

Comment (cionescu) [inline app/models/answer.rb]: Same here
`where.not(questions: { type: "CodeQuestion" })`

File: app/models/answer.rb [hunk 2]
```diff
@@ -24,4 +24,7 @@ class Answer < ApplicationRecord
...
+scope :application_answers, -> {joins(:question).where('questions.type != ?', 'CodeQuestion')}
```
Comment (cionescu) [inline app/models/answer.rb]: Same here
`where.not(questions: { type: "CodeQuestion" })`

File: app/models/question.rb [hunk 3]
```diff
@@ -27,12 +27,13 @@ class Question < ApplicationRecord
   scope :boolean_questions, -> { where(type: 'BooleanQuestion') }
   scope :select_questions, -> { where(type: 'SelectQuestion') }
   scope :text_questions, -> { where(type: 'TextQuestion') }
-
+  scope :code_questions, -> { where(type: 'CodeQuestion') }
+  scope :sign_up_questions, -> { where(programme_application_id: nil) }
```
Comment (cionescu) [inline app/models/question.rb]: I thought we agreed these questions would have a separate question type

File: app/views/applicants/show.html.haml [hunk 4]
```diff
@@ -1,104 +1,132 @@
...
+                    .card-image.circle
+                      = image_tag avatar_url(current_user), class: 'circle'
+                .col.s8
+                  .card-content
+                    %span.card-title= current_user.display_name
+                    %p= current_user.email
+                    %p= current_user.phone_number
+                    %p= current_user.name
+                    - if @account.present?
+                      - {github: 'github', linkedin: 'linkedin', twitter: 'twitter', website: 'globe'}.each do |k,v|
```
Comment (cionescu) [inline app/views/applicants/show.html.haml]: 😛 Please move this hash to a helper method 😛

File: app/views/applicants/show.html.haml [hunk 5]
```diff
@@ -1,104 +1,132 @@
...
+                        = link_to @account.send(k), target: '_blank' do
```
Comment (cionescu) [inline app/views/applicants/show.html.haml]: Send is a really dangerous method to use (and the code is not at all intuitive). You could use an internal AR representation such as `@account.attributes["github"]`

File: db/migrate/20160809042839_add_name_to_users.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class AddNameToUsers < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :name, :string
```
Comment (cionescu) [inline db/migrate/20160809042839_add_name_to_users.rb]: Can you add a `default: ""` which will help you avoid doing things like `name&.strip || "blah"`. The name should never be nil is my point

File: app/models/question.rb (hunk 3 above)
Comment (faethonm) [inline app/models/question.rb]: I dont think its needed

**Pull request merged**